src/Microsoft.DotNet.Cli.Utils/Env.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    internal static class Env
+    {
+        private static IEnumerable<string> _searchPaths;
+        private static IEnumerable<string> _executableExtensions;
+
+        public static IEnumerable<string> ExecutableExtensions
+        {
+            get
+            {
+                if (_executableExtensions == null)
+                {
+
+                    _executableExtensions = RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
+                        ? Environment.GetEnvironmentVariable("PATHEXT").Split(';').Select(e => e.ToLower())
+                        : new [] { string.Empty };
+                }
+
+                return _executableExtensions;;
+            }
+        } 
+
+        private static IEnumerable<string> SearchPaths 
+        {
+            get
+            {
+                if (_searchPaths == null)
+                {
+                    var searchPaths = new List<string> {AppContext.BaseDirectory};
+
+                    searchPaths.AddRange(Environment.GetEnvironmentVariable("PATH").Split(Path.PathSeparator));
```

Comment:
On windows it is ; and its : on *nix. FYI this is the separator used in the PATH env var and not the directory separator char.